lib/model.js

```diff
@@ -1341,19 +1341,22 @@ Model.init = function init(callback) {
 
   this.schema.emit('init', this);
 
-  if (this.$init != null) {
+  const autoCreate = utils.getOption('autoCreate',
+    this.schema.options, this.db.config, this.db.base.options);
+
+  if (this.$init != null && this.$autoCreate === autoCreate) {
```

Comment:
The problem with this approach is that you may have multiple `$init`'s going at the same time. `Model.init(); Model.schema.options.autoCreate = false; Model.init();`. The purpose of `$init` is to avoid that.

I think a better approach would be to move the `utils.getOption()` call for `autoCreate` into the `_createCollection()` function itself. And put `_createCollection()` behind buffering like how `_wrapConnHelper()` works.

Reply:
Thanks for the feedback 💪 I'll work on this